cosmos/x/evm/plugins/txpool/mempool/mempool.go

```diff
@@ -43,6 +39,11 @@ type EthTxPool struct {
 	// later
 	ethTxCache map[common.Hash]*coretypes.Transaction
 
+	// nonceToHash maps a nonce to the hash of the transaction that was added to the mempool with
+	// that nonce. This is used to retrieve the hash of a transaction that was added to the mempool
+	// by nonce.
+	nonceToHash map[uint64]common.Hash
```

Comment:
this needs to be for each sender addr: `map[common.Address]map[uint64]common.Hash`